producers/user_data_wh/data_producer.py

```diff
@@ -0,0 +1,187 @@
+                "userType": user_type,
+                "username": username,
+                "email": email,
+                "password": password,
+            }
+            if test:
+                self.records_produced.append(user_dict)
+            yield user_dict
+
+    def member_generator(self, test=False):
+        """ Create regular member n_record generator using faker library """
+        for _ in range(self.n_records):
+            fake = Faker()
+            name = fake.name().split()
+            email = "".join(name).lower() + "@gmail.com"
+            username = "".join(name).lower()
+            password = fake.password(length=13, special_chars=False)
+            firstname = name[0].strip().lower()
+            lastname = name[1].strip().lower()
+            dob = fake.date_of_birth(minimum_age=18).strftime("%Y-%m-%d")
+            ssn = fake.ssn()
+            full_address = fake.address().split()
+            address = " ".join(full_address[: -3 or None])
+            city = fake.city().replace(" ", "")
+            state = full_address[-2]
+            zipcode = full_address[-1]
+
+            member_dict = {
+                "email": email,
+                "username": username,
+                "password": password,
+                "firstName": re.sub(r"\W+", "", firstname),
+                "lastName": re.sub(r"\W+", "", lastname),
+                "phone": phn(),
+                "dateOfBirth": dob,
+                "socialSecurityNumber": ssn,
+                "addressLine1": re.sub(r"\W+", "", address),
+                "city": city.strip(),
+                "state": state,
+                "zipcode": zipcode,
+            }
+
+            if test:
+                self.records_produced.append(member_dict)
+            yield member_dict
+
+    def records_insert(self, generator):
+        """ depending on data type insert data produced from generator to route """
+        if self.data_type == "user":
+            route = self.create_user_route
+        else:
+            route = self.create_member_route
+
+        headers = {"Content-Type": "application/json"}
+
+        url = self.base_url + route
+
+        try:
+            while True:
+                data = next(generator)
+                if data is not None:
+                    resp = requests.request(
+                        "POST", url, headers=headers, json=data, verify=self.crt_path
+                    )
+                    if resp.status_code != 201:
+                        raise requests.exceptions.HTTPError()
+                    print(f"INSERTED {data['username']} as {self.data_type}")
+                    continue
+                break
+        except requests.exceptions.HTTPError as err:
+            print(err, resp.status_code)
+            if resp.status_code == 400:
+                return True
+            return False
+        return True
+
+
+def phn():
+    """ Nice XXX-XXX-XXXX phone number gen I lifted off of stack overflow """
```

Comment:
might want to link the StackOverflow question that you referenced here

Reply:
yeah gotcha just added it